From 247114342fd47d24152e7bfd05abf847fdf02a31 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 22 Jan 2025 15:13:52 +0100
Subject: [PATCH] drm/stm: ltdc: ckeck pm_runtime status

Check the pm_runtime status to disable clocks when the driver is removed.

Change-Id: Ib935972c1c930be36b983d7819ae312f1c4be7c9
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/437318
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index e55fe70685b3..ff76e618b1d2 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -2631,6 +2631,9 @@ void ltdc_unload(struct drm_device *ddev)
 
 	DRM_DEBUG_DRIVER("\n");
 
+	if (pm_runtime_active(ddev->dev))
+		pm_runtime_put_sync_suspend(ddev->dev);
+
 	stm32_firewall_release_access(fwl);
 
 	nb_endpoints = of_graph_get_endpoint_count(dev->of_node);
-- 
2.47.2

